backend/application/syson-application/src/test/java/org/eclipse/syson/application/controllers/diagrams/general/view/GVAddExistingElementsTests.java

```diff
@@ -179,78 +182,114 @@ public void addExistingElementsRecursiveOnDiagram() {
                     // .anyMatch(n -> n.getStyle() instanceof ImageNodeStyle imageStyle &&
                     // Objects.equals(imageStyle.getImageURL(), "images/start_action.svg")
                     // && Objects.equals("start", n.getTargetObjectLabel()));
 
-                    var optAction2Node = action1ActionFlowCompartment.get().getChildNodes().stream()
-                            .filter(n -> Objects.equals(n.getTargetObjectLabel(), ACTION2))
-                            .findFirst();
-                    assertThat(optAction2Node).isPresent();
-                    var action2Node = optAction2Node.get();
-                    var action2ActionsCompartment = this.getCompartment(action2Node, "actions");
-                    assertThat(action2ActionsCompartment)
-                            .as(ACTION2 + " should contain an actions compartment")
-                            .isPresent();
-                    assertThat(action2ActionsCompartment.get().getChildNodes())
-                            .as(ACTION2 + " actions compartment should contain 1 child")
-                            .hasSize(1)
-                            .as(ACTION2 + " actions compartment should contain " + ACTION3)
-                            .anyMatch(n -> Objects.equals(n.getTargetObjectLabel(), ACTION3));
-
-                    var action2ActionFlowCompartment = this.getCompartment(action2Node, "action flow");
-                    assertThat(action2ActionFlowCompartment)
-                            .as(ACTION2 + " should contain an action flow compartment")
-                            .isPresent();
-                    assertThat(action2ActionFlowCompartment.get().getChildNodes())
-                            .as(ACTION2 + " action flow compartment should contain 1 child")
-                            .hasSize(1)
-                            .as(ACTION2 + " action flow compartment should contain " + ACTION3)
-                            .anyMatch(n -> Objects.equals(n.getTargetObjectLabel(), ACTION3));
+                    checkAction2(action1ActionFlowCompartment);
+
+                    checkRequirementUsage(newDiagram);
 
                 }, () -> fail("Missing diagram"));
         this.verifier.consumeNextWith(updatedDiagramConsumer);
     }
 
+    private static void checkPackageNode(Diagram newDiagram) {
+        var optPackageNode = newDiagram.getNodes().stream()
+                .filter(n -> Objects.equals(n.getTargetObjectLabel(), PACKAGE1))
+                .findFirst();
+        assertThat(optPackageNode).isPresent();
+        assertThat(optPackageNode.get().getChildNodes())
+                .as("Node " + PACKAGE1 + " should contain 1 child")
+                .hasSize(1)
+                .as("Node " + ATTRIBUTE_DEFINITION + " should exist inside " + PACKAGE1)
+                .anyMatch(n -> Objects.equals(n.getTargetObjectLabel(), ATTRIBUTE_DEFINITION));
+    }
+
+    private void checkAction2(Optional<Node> action1ActionFlowCompartment) {
```

Comment:
I don't think Optional is a good idea for a parameter.
You should probably add a check on action1ActionFlowCompartment before calling this method.

Reply:
You are right.  I simply extract methods du to checkstyle constraint, I should have done it manually.